Add a new page dedicated to the asChild prop (#628)
radix-ui/website: data/primitives/guides/composition.mdx

In reality, you will very rarely modify the underlying DOM element directly like we've seen above. Instead you will most likely want to use your own React components. This is especially true for most `Trigger` parts, as you usually want to compose the functionality with your own button or link components from your design system.

## Composing with your own React components

This works exactly the same as above, you pass `asChild` to the part and then wrap your own component with it.
However, there are a few gotchas to be aware of.

### Your component needs to spread props

When Radix clones your component, it will pass certain props and event handlers to it to make it functional and accessible. If your component doesn't support those props, then it will break.

This is done by spreading all of the props onto the underlying DOM node.

```jsx line=5
// before
const MyButton = () => <button />;

// after
const MyButton = (props) => <button {...__props__} />;
```

> We recommend always doing it so that you are not concerned with implementation details (ie. which props/events to accept). This is also generally good practice anyway for "leaf" components.

Reviewer: ```suggestion
We recommend always doing this so that you are not concerned with implementation details (ie. which props/events to accept). We find this is good practice for "leaf" components in general.
```

> Similarly to when changing the underlying element type directly, it is your responsibility to ensure the element type rendered by your custom component is compatible in terms of accessibility and functionality.

### Your component needs to forward ref

Additionally, Radix will sometimes need to attach a `ref` to your component (for example to measure its size). If your component doesn't accept a `ref`, then it will break.

This is done using `React.forwardRef` (read more on [react.dev](https://react.dev/reference/react/forwardRef)).

```jsx line=5-7
// before
const MyButton = (props) => <button {...props} />;

// after
const MyButton = __React.forwardRef__((props, __forwardedRef__) => (
  <button {...props} __ref__={__forwardedRef__} />
));
```

Whilst this isn't necessary for **all** parts, we recommend always doing it so that you are not concerned with implementation details. This is also generally good practice anyway for leaf components.

## Composing multiple primitives

> `asChild` can be used as deeply as you need to. This means it is a great way to compose multiple primitive's behaviour together.

Reviewer: ```suggestion
`asChild` can be used as deeply as you need to. This means it is a great way to compose multiple primitive's behavior together.
```
